File: 0332-reconstruct-itinerary/0332-reconstruct-itinerary.java
class Solution {
    public List<String> findItinerary(List<List<String>> tickets) {
        LinkedList<String> ans =new LinkedList<>(); 
        HashMap<String,PriorityQueue<String>> adj=new HashMap<>();
        for(List<String> ticket:tickets){
            String u=ticket.get(0);
            String v=ticket.get(1);

            adj.putIfAbsent(u,new PriorityQueue<String>());
            adj.get(u).add(v);
        }
        dfs("JFK",adj,ans);
        return ans;
    }
    public void dfs(String src,HashMap<String,PriorityQueue<String>> adj,LinkedList<String> ans){
        PriorityQueue<String> temp=adj.get(src);

        while(temp!=null && temp.size()>0){
            String s=temp.poll();
            dfs(s,adj,ans);
        }
        ans.addFirst(src);
        return;
    }
}

File: 0332-reconstruct-itinerary/0332-reconstruct-itinerary.js
var findItinerary = function(tickets) {
    var map = {};
    var res = [];
    for(var i=0; i<tickets.length; i++) {
        var dep = tickets[i][0];
        var des = tickets[i][1];
        if(map[dep]) {
            map[dep].push(des);
        } else {
            map[dep] = [des];
        }
    }
    for(let loc in map) {
        map[loc].sort();
    }  
    var dfs = function(node) {
        var des = map[node];
        while(des && des.length>0) {
            dfs(des.shift());
        }
        res.push(node);
    }
    dfs('JFK');
    return res.reverse();
};